WordPress/src/main/java/org/wordpress/android/viewmodel/posts/PostListViewModel.kt

```diff
@@ -179,30 +181,100 @@ class PostListViewModel @Inject constructor(
         }
         result
     }
-    val emptyViewState: LiveData<PostListEmptyViewState> by lazy {
-        val result = MediatorLiveData<PostListEmptyViewState>()
+    val emptyViewState: LiveData<PostListEmptyUiState> by lazy {
+        val result = MediatorLiveData<PostListEmptyUiState>()
         val update = {
-            val error = pagedListWrapper.listError.value
-            if (pagedListWrapper.isEmpty.value != false) {
-                if (error != null) {
-                    if (error.type == ListErrorType.PERMISSION_ERROR) {
-                        PostListEmptyViewState.PERMISSION_ERROR
-                    } else PostListEmptyViewState.REFRESH_ERROR
-                } else if (pagedListWrapper.isFetchingFirstPage.value == true) {
-                    PostListEmptyViewState.LOADING
-                } else {
-                    PostListEmptyViewState.EMPTY_LIST
-                }
-            } else {
-                PostListEmptyViewState.HIDDEN_LIST
-            }
+            createEmptyUiState()
         }
         result.addSource(pagedListWrapper.isEmpty) { result.value = update() }
         result.addSource(pagedListWrapper.isFetchingFirstPage) { result.value = update() }
         result.addSource(pagedListWrapper.listError) { result.value = update() }
         result
     }
 
+    private fun createEmptyUiState(): PostListEmptyUiState {
```

Comment:
I think it'd be better to take a `PagedListWrapper` instance in this method instead of accessing the global value. Just looking at this method alone, it's not clear that it's safe to read the value of `PagedListWrapper`s state values directly. I'd even take it one step further and pass whatever this method needs, such as `isEmpty`, `listError` etc, and not expose the `PagedListWrapper` at all. It really doesn't matter if we use a `PagedListWrapper` or something else to generate this empty ui state.

We might also consider moving all this empty list logic to a helper somewhere? Maybe we could add it all to a file called `PostListEmptyUiState` and make the helpers just `internal` functions. It's not like they are testable as is, since they are private, so we wouldn't really lose anything if I am not missing anything.

Reply:
>I think it'd be better to take a `PagedListWrapper` instance in this method instead of accessing the global value. Just looking at this method alone, it's not clear that it's safe to read the value of `PagedListWrapper`s state values directly. I'd even take it one step further and pass whatever this method needs, such as `isEmpty`, `listError` etc, and not expose the `PagedListWrapper` at all. It really doesn't matter if we use a `PagedListWrapper` or something else to generate this empty ui state.

Fixed in https://github.com/wordpress-mobile/WordPress-Android/pull/9287/commits/d19f6131895b51e69cc26d8f10e12a990b41141f.

>We might also consider moving all this empty list logic to a helper somewhere? Maybe we could add it all to a file called `PostListEmptyUiState` and make the helpers just `internal` functions. It's not like they are testable as is, since they are private, so we wouldn't really lose anything if I am not missing anything.

I feel like everyone will look for the UiState in the ViewModel - especially it's in there in all the other ViewModels.